homeassistant/components/transmission/__init__.py

```diff
@@ -138,17 +145,19 @@ async def get_api(hass, entry):
 class TransmissionClient:
     """Transmission Client Object."""
 
-    def __init__(self, hass, config_entry):
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         """Initialize the Transmission RPC API."""
         self.hass = hass
         self.config_entry = config_entry
-        self.tm_api: transmissionrpc.Client = None
-        self._tm_data: TransmissionData = None
+        self.tm_api: transmissionrpc.Client | None = None
+        self._tm_data: TransmissionData | None = None
         self.unsub_timer = None
 
     @property
     def api(self) -> TransmissionData:
         """Return the TransmissionData object."""
+        if not self._tm_data:
+            raise HomeAssistantError("Transmission client not initialised")
```

Comment:
There should not be any case where the integration is initialized and `self._tm_data` remains None. If this check is from mypy then I suggest you do the following:
- Get `tm_api` under `async_setup_entry`. This should fail the integration immediately if it raises.
- pass `tm_api` to `TransmissionClient` and store it directly in `self.tm_api` (so it will never be None).
- Initialize `self._tm_data` directly `self._tm_data = TransmissionData(hass, config_entry, tm_api)`